src/main/java/com/github/jcustenborder/kafka/connect/snmp/SnmpTrapSourceConnectorConfig.java

```diff
@@ -72,17 +118,42 @@ public SnmpTrapSourceConnectorConfig(Map<String, String> parsedConfig) {
+        .stream().map((s) -> AuthenticationProtocol.valueOf(s.toUpperCase()))
+        .collect(Collectors.toSet());
+    this.privacyProtocols = this.getList(PRIVACY_PROTOCOLS)
+        .stream().map((s) -> PrivacyProtocol.valueOf(s.toUpperCase()))
+        .collect(Collectors.toSet());
+    this.username = this.getString(USM_USERNAME);
+    this.authenticationPassphrase = this.getString(USM_AUTHENTICATION_PASSPHRASE);
+    this.privacyPassphrase = this.getString(USM_PRIVACY_PASSPHRASE);
+    this.authenticationProtocol =  AuthenticationProtocol.valueOf(this.getString(USM_AUTHENTICATION_PROTOCOL).toUpperCase());
+    this.privacyProtocol = PrivacyProtocol.valueOf(this.getString(USM_PRIVACY_PROTOCOL).toUpperCase());
   }
 
   public static ConfigDef conf() {
+    String[] authProtocols = Arrays.stream(AuthenticationProtocol.values()).map(Enum::toString).toArray(String[]::new);
+    String[] privProtocols = Arrays.stream(PrivacyProtocol.values()).map(Enum::toString).toArray(String[]::new);
+
     return new ConfigDef()
-            .define(TOPIC_CONF, Type.STRING, Importance.HIGH, TOPIC_DOC)
-            .define(LISTEN_ADDRESS_CONF, Type.STRING, LISTEN_ADDRESS_DEFAULT, Importance.LOW, LISTEN_ADDRESS_DOC)
-            .define(LISTEN_PORT_CONF, Type.INT, LISTEN_PORT_DEFAULT, Validators.validPort(161, 65535), Importance.LOW, LISTEN_PORT_DOC)
-            .define(LISTEN_PROTOCOL_CONF, Type.STRING, LISTEN_PROTOCOL_DEFAULT, ConfigDef.ValidString.in("UDP", "TCP"), Importance.LOW, LISTEN_PROTOCOL_DOC)
-            .define(DISPATCHER_THREAD_POOL_SIZE_CONF, Type.INT, DISPATCHER_THREAD_POOL_SIZE_DEFAULT, ConfigDef.Range.between(1, 100), Importance.LOW, DISPATCHER_THREAD_POOL_SIZE_DOC)
-            .define(BATCH_SIZE_CONF, Type.INT, BATCH_SIZE_DEFAULT, ConfigDef.Range.between(10, Integer.MAX_VALUE), Importance.MEDIUM, BATCH_SIZE_DOC)
-            .define(POLL_BACKOFF_MS_CONF, Type.LONG, POLL_BACKOFF_MS_DEFAULT, ConfigDef.Range.between(10, Integer.MAX_VALUE), Importance.MEDIUM, POLL_BACKOFF_MS_DOC);
+        .define(TOPIC_CONF, Type.STRING, Importance.HIGH, TOPIC_DOC)
+        .define(LISTEN_ADDRESS_CONF, Type.STRING, LISTEN_ADDRESS_DEFAULT, Importance.LOW, LISTEN_ADDRESS_DOC)
+        .define(LISTEN_PORT_CONF, Type.INT, LISTEN_PORT_DEFAULT, Validators.validPort(161, 65535), Importance.LOW, LISTEN_PORT_DOC)
+        .define(LISTEN_PROTOCOL_CONF, Type.STRING, LISTEN_PROTOCOL_DEFAULT, ConfigDef.ValidString.in("UDP", "TCP"), Importance.LOW, LISTEN_PROTOCOL_DOC)
+        .define(DISPATCHER_THREAD_POOL_SIZE_CONF, Type.INT, DISPATCHER_THREAD_POOL_SIZE_DEFAULT, ConfigDef.Range.between(1, 100), Importance.LOW, DISPATCHER_THREAD_POOL_SIZE_DOC)
+        .define(BATCH_SIZE_CONF, Type.INT, BATCH_SIZE_DEFAULT, ConfigDef.Range.between(10, Integer.MAX_VALUE), Importance.MEDIUM, BATCH_SIZE_DOC)
+        .define(POLL_BACKOFF_MS_CONF, Type.LONG, POLL_BACKOFF_MS_DEFAULT, ConfigDef.Range.between(10, Integer.MAX_VALUE), Importance.MEDIUM, POLL_BACKOFF_MS_DOC)
```

Comment:
`POLL_BACKOFF_MS` high end is quite high with `Integer.MAX_VALUE` - maybe decrease that to shorter?

Reply:
I think this is fine since it is just for validator and it would be artificial upper limit, it will default to 250 anyway.